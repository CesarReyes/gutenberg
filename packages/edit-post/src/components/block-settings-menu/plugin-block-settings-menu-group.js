/**
 * External dependencies
 */
import { filter, isEmpty, map } from 'lodash';

/**
 * WordPress dependencies
 */
import { createSlotFill } from '@wordpress/components';
import { Fragment } from '@wordpress/element';
import { withSelect } from '@wordpress/data';

const { Fill: PluginBlockSettingsMenuGroup, Slot } = createSlotFill( 'PluginBlockSettingsMenuGroup' );

export const PluginBlockSettingsMenuGroupSlot = ( { fillProps, selectedBlocks } ) => {
	const blockNames = map( filter( selectedBlocks, ( block ) => ( block !== null ) && block.name ), ( block ) => block.name );
	if ( blockNames.length === 0 ) {
		return null;
	}

	return (
		<Slot fillProps={ { ...fillProps, selectedBlocks: blockNames } } >
			{ ( fills ) => ! isEmpty( fills ) && (
				<Fragment>
					<div className="editor-block-settings-menu__separator" />
					{ fills }
				</Fragment>
			) }
		</Slot>
	);
};

PluginBlockSettingsMenuGroup.Slot = withSelect( ( select, { fillProps: { clientIds } } ) => ( {
	selectedBlocks: select( 'core/editor' ).getBlocksByClientId( clientIds ),
} ) )( PluginBlockSettingsMenuGroupSlot );

export default PluginBlockSettingsMenuGroup;
