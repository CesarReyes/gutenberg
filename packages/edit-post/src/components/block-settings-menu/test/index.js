/**
 * External dependencies.
 */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Internal dependencies.
 */
import { PluginBlockSettingsMenuGroupSlot } from '../plugin-block-settings-menu-group';

describe( 'PluginBlockSettingsMenuGroupSlot', () => {
	test( 'should render normally if block selected has name property', () => {
		const renderer = new ShallowRenderer();
		renderer.render(
			<PluginBlockSettingsMenuGroupSlot selectedBlocks={ [ { name: 'paragraph' } ] } />
		);

		expect( renderer.getRenderOutput() ).toMatchSnapshot();
	} );

	test( 'should be hidden if no block is selected', () => {
		const renderer = new ShallowRenderer();
		renderer.render(
			<PluginBlockSettingsMenuGroupSlot selectedBlocks={ [ null ] } />
		);

		expect( renderer.getRenderOutput() ).toBeNull();
	} );

	test( 'should be hidden if block selected has no name declared', () => {
		const renderer = new ShallowRenderer();
		renderer.render(
			<PluginBlockSettingsMenuGroupSlot selectedBlocks={ [ {} ] } />
		);

		expect( renderer.getRenderOutput() ).toBeNull();
	} );
} );
