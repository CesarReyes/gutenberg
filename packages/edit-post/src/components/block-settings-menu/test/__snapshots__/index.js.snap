// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginBlockSettingsMenuGroupSlot should render normally if block selected has name property 1`] = `
<PluginBlockSettingsMenuGroupSlot
  fillProps={
    Object {
      "selectedBlocks": Array [
        "paragraph",
      ],
    }
  }
>
  [Function]
</PluginBlockSettingsMenuGroupSlot>
`;
